### ISSUE TITLE
AWS deploy problem when using IAM role

### ISSUE THREAD

**Issue by dwhitena** [pachyderm/pachyderm#2522]
One of our users reported this in Slack. I'm going to try to reproduce today, but wanted to capture this here, as it seems like something weird is going on.

## Environment

- AWS w/out Cloudfront
- IAM role created according to https://github.com/kubernetes/kops/blob/master/docs/aws.md
- Pachyderm 1.6.3 or 1.6.4
- Kubernetes:

    ```
    $ kubectl version
    Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-09T07:27:47Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
    Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.10", GitCommit:"bebdeb749f1fa3da9e1312c4b08e439c404b3136", GitTreeState:"clean", BuildDate:"2017-11-03T16:31:49Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
    ```

## Behavior

When deploying to AWS with the [manual deploy](http://pachyderm.readthedocs.io/en/latest/deployment/amazon_web_services.html#manual-pachyderm-deploy) and using an IAM role via the `--iam-role` flag, `pachd` get's stuck in `CrashLoopBackoff` with the following error:

```sh
$ kubectl logs po/pachd-1269558310-lljzw
time="2017-11-16T15:41:33Z" level=info msg="validating kubernetes access returned no errors"
time="2017-11-16T15:41:33Z" level=info msg="AWS deployed with cloudfront distribution at
"
time="2017-11-16T15:41:33Z" level=info msg="Launching PPS master process"
storage is unable to discern NotExist errors, "AccessDenied: Access Denied
    status code: 403, request id: 613CB7E9A9263C79, host id: yQa3wa1XSUIu0WgQtmu0Zvb7F/doxT7YHI5R1A0b0dkbAlIhR4EGIhZZkxAWwXUgtjRAEz9FWHc=" should count as NotExist
```

At first glance, this seems to be related to: https://github.com/pachyderm/pachyderm/issues/1917, but note that the user is not using Cloudfront.

## Reproducing 

Again, I haven't been able to try this yet, but according to the steps I went through with the user:

1. Create IAM role based on https://github.com/kubernetes/kops/blob/master/docs/aws.md
2. Follow the AWS manual deploy docs using the IAM role, just as they are written.
3. Watch for the `CrashLoopBackoff` on the `pachd` pod
4. Check the logs for the `pachd` pod with `kubectl logs`

## Notes

@sjezewski Maybe you have some thoughts here?

**Comment by scheidec** [comment]
I was running into this issue yesterday when trying the manual deploy with the IAM role, but I tried again this morning using the one-shot script (static credentials) without using an IAM role and it appears there are no `pachd` issues that way.

**Comment by dwhitena** [comment]
Cool. Thanks for the details @scheidec! So this appears to be an apparent issue related to the way we are handling IAM roles. We will definitely try to reproduce.

**Comment by sjezewski** [comment]
Yea, so what's happening here is that pachd will use the ec2's instance IAM role by default.

(This is true no matter what you've set the `--iam-role` flag to, which is why our docs need updating and this is confusing: https://github.com/pachyderm/pachyderm/issues/2680)

That error message happens when that EC2 iam role does not have access to the s3 bucket specified. I saw this error when explicitly testing this out yesterday. I expect that error when the role doesn't have access. Once I added s3 access to that role, and restarted pachd, the error went away.

Hopefully that makes this a bit clearer. We have some updating to do on our docs to make this more transparent.
